### PR TITLE
fix: show ErrorSnackMessage in EventDetailsCoordinatorLayout

### LINKED ISSUE #1553: Improper Placement of "error fetching social links" snackbar
**Describe the feature you'd like**
Due to Internet issues (Suppose user open the vent details , suddenly internet access is gone) , when there is error in fetching similar events or social links. Snackbar should be shown to users .` But currently user has to scroll to see that snackbar`

**Screenshots**

**Would you like to work on the issue?**

- [ :heavy_check_mark:  ] Yes
- [ ] No
- Other:

### PR DESCRIPTION
**Pull request by theLimitBreaker**
Fixes #1553 

Changes:
 Change the targetView of SnackBar to `EventDetailsCoordinatorLayout` to show the snackMessage without Scrolling

Screenshots for the change:

### CODE REVIEW

**Comment by theLimitBreaker** [comment]
@ShridharGoel  @liveHarshit  please review

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt [hunk 1]
```diff
@@ -18,34 +18,14 @@ import androidx.fragment.app.Fragment
...
-import kotlinx.android.synthetic.main.content_event.view.organizerContainer
-import kotlinx.android.synthetic.main.fragment_event.view.buttonTickets
-import kotlinx.android.synthetic.main.fragment_event.view.eventErrorCard
-import kotlinx.android.synthetic.main.fragment_event.view.container
+import kotlinx.android.synthetic.main.content_event.*
+import kotlinx.android.synthetic.main.content_event.view.*
 import kotlinx.android.synthetic.main.content_fetching_event_error.view.retry
+import kotlinx.android.synthetic.main.fragment_event.*
+import kotlinx.android.synthetic.main.fragment_event.view.*
+import kotlinx.android.synthetic.main.fragment_event.view.eventErrorCard
```
Comment (ShridharGoel) [inline app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: Remove wildcard imports

File: app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsFragment.kt [hunk 2]
```diff
@@ -9,25 +9,20 @@ import androidx.core.view.isVisible
...
-import com.google.android.material.snackbar.Snackbar
-import kotlinx.android.synthetic.main.fragment_similar_events.moreLikeThis
-import kotlinx.android.synthetic.main.fragment_similar_events.progressBar
-import kotlinx.android.synthetic.main.fragment_similar_events.similarEventsDivider
-import kotlinx.android.synthetic.main.fragment_similar_events.similarEventsRecycler
-import kotlinx.android.synthetic.main.fragment_similar_events.view.similarEventsRecycler
-import kotlinx.android.synthetic.main.fragment_similar_events.view.similarEventsCoordinatorLayout
+import androidx.recyclerview.widget.LinearLayoutManager
+import kotlinx.android.synthetic.main.fragment_similar_events.*
+import kotlinx.android.synthetic.main.fragment_similar_events.view.*
```
Comment (ShridharGoel) [inline app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsFragment.kt]: Wildcard imports

File: app/src/main/java/org/fossasia/openevent/general/social/SocialLinksFragment.kt [hunk 3]
```diff
@@ -9,14 +9,8 @@ import androidx.core.view.isVisible
...
-import com.google.android.material.snackbar.Snackbar
-import kotlinx.android.synthetic.main.fragment_social_links.eventHostDetails
-import kotlinx.android.synthetic.main.fragment_social_links.socialLinksRecycler
-import kotlinx.android.synthetic.main.fragment_social_links.socialLinksCoordinatorLayout
-import kotlinx.android.synthetic.main.fragment_social_links.view.progressBarSocial
-import kotlinx.android.synthetic.main.fragment_social_links.view.socialLinksRecycler
-import kotlinx.android.synthetic.main.fragment_social_links.view.socialLinkReload
-import kotlinx.android.synthetic.main.fragment_social_links.view.socialNoInternet
+import kotlinx.android.synthetic.main.fragment_social_links.*
+import kotlinx.android.synthetic.main.fragment_social_links.view.*
```
Comment (ShridharGoel) [inline app/src/main/java/org/fossasia/openevent/general/social/SocialLinksFragment.kt]: Wildcard imports

Comment (theLimitBreaker) [reply app/src/main/java/org/fossasia/openevent/general/social/SocialLinksFragment.kt]: It passed spotless check so thought its fine. Doing it

Comment (theLimitBreaker) [reply app/src/main/java/org/fossasia/openevent/general/social/SocialLinksFragment.kt]: resolved

**Comment by theLimitBreaker** [comment]
@liveHarshit  @ShridharGoel  could you review and help with codacy issue

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt [hunk 4]
```diff
@@ -56,13 +58,13 @@ import org.fossasia.openevent.general.ticket.TicketsFragmentArgs
...
 import org.fossasia.openevent.general.utils.Utils.getAnimSlide
 import org.fossasia.openevent.general.utils.Utils.requireDrawable
+import org.fossasia.openevent.general.utils.Utils.setToolbar
 import org.fossasia.openevent.general.utils.extensions.nonNull
 import org.fossasia.openevent.general.utils.nullToEmpty
 import org.fossasia.openevent.general.utils.stripHtml
 import org.koin.androidx.viewmodel.ext.android.viewModel
 import timber.log.Timber
 import java.util.Currency
-import org.fossasia.openevent.general.utils.Utils.setToolbar
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: Unnecessary changes

File: app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsFragment.kt [hunk 5]
```diff
@@ -9,25 +9,23 @@ import androidx.core.view.isVisible
...
 import kotlinx.android.synthetic.main.fragment_similar_events.moreLikeThis
 import kotlinx.android.synthetic.main.fragment_similar_events.progressBar
 import kotlinx.android.synthetic.main.fragment_similar_events.similarEventsDivider
 import kotlinx.android.synthetic.main.fragment_similar_events.similarEventsRecycler
 import kotlinx.android.synthetic.main.fragment_similar_events.view.similarEventsRecycler
-import kotlinx.android.synthetic.main.fragment_similar_events.view.similarEventsCoordinatorLayout
 import org.fossasia.openevent.general.R
+import org.fossasia.openevent.general.common.EventClickListener
+import org.fossasia.openevent.general.common.FavoriteFabClickListener
+import org.fossasia.openevent.general.common.ShareFabClickListener
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsFragment.kt]: same

File: app/src/main/java/org/fossasia/openevent/general/social/SocialLinksFragment.kt [hunk 6]
```diff
@@ -9,13 +9,11 @@ import androidx.core.view.isVisible
...
 import androidx.lifecycle.Observer
 import androidx.recyclerview.widget.LinearLayoutManager
-import com.google.android.material.snackbar.Snackbar
 import kotlinx.android.synthetic.main.fragment_social_links.eventHostDetails
 import kotlinx.android.synthetic.main.fragment_social_links.socialLinksRecycler
-import kotlinx.android.synthetic.main.fragment_social_links.socialLinksCoordinatorLayout
 import kotlinx.android.synthetic.main.fragment_social_links.view.progressBarSocial
-import kotlinx.android.synthetic.main.fragment_social_links.view.socialLinksRecycler
 import kotlinx.android.synthetic.main.fragment_social_links.view.socialLinkReload
+import kotlinx.android.synthetic.main.fragment_social_links.view.socialLinksRecycler
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/social/SocialLinksFragment.kt]: same

File: app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsFragment.kt (hunk 5 above)
Comment (theLimitBreaker) [reply app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsFragment.kt]: You mean by order of imports

File: app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt (hunk 4 above)
Comment (iamareebjamal) [reply app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt]: Rearrangement of imports is fine

**Comment by aggarwalpulkit596** [comment]
resolve conflicts

**Comment by theLimitBreaker** [comment]
@anhanh11001 @angmas1 Peer review please

File: app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsFragment.kt [hunk 7]
```diff
@@ -8,25 +8,23 @@ import androidx.core.view.isVisible
...
 import androidx.navigation.fragment.navArgs
+import androidx.recyclerview.widget.LinearLayoutManager
 import kotlinx.android.synthetic.main.fragment_similar_events.moreLikeThis
 import kotlinx.android.synthetic.main.fragment_similar_events.progressBar
 import kotlinx.android.synthetic.main.fragment_similar_events.similarEventsDivider
 import kotlinx.android.synthetic.main.fragment_similar_events.similarEventsRecycler
 import kotlinx.android.synthetic.main.fragment_similar_events.view.similarEventsRecycler
-import kotlinx.android.synthetic.main.fragment_similar_events.view.similarEventsCoordinatorLayout
 import org.fossasia.openevent.general.R
+import org.fossasia.openevent.general.common.EventClickListener
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsFragment.kt]: Unrelated changes in imports.

**Comment by theLimitBreaker** [comment]
@liveHarshit @iamareebjamal updated

File: app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsFragment.kt (hunk 7 above)
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsFragment.kt]: Not updated!

Comment (theLimitBreaker) [reply app/src/main/java/org/fossasia/openevent/general/event/topic/SimilarEventsFragment.kt]: updated

**Pull request merged**